dashboard/src/components/api/che-workspace.factory.ts

```diff
@@ -646,4 +652,22 @@ export class CheWorkspace {
     this.workspacesById.set(workspace.id, workspace);
     this.startUpdateWorkspaceStatus(workspace.id);
   }
+
+  private connectWsMasterApi($location: ng.ILocationService, proxySettings : string, devmode: boolean): void {
```

Comment:
the pattern is : new CheJsonRPCMasterAPI(websocketClient)
and then : rpcMasterAPI.connect(endpoint)
IMHO we're mixing two patterns because object is not useable until we're connected.
could we have it in a more "factory pattern" ?
 like we get a new dedicated object each time we connect on an endpoint (and it may be pooled as well)

factory = new RPCJsonFactory(configuration)
factory.get(endpoint)  --> return instance of CheJsonRpcMasterApi